src/transformStateWithClones.js

```diff
@@ -6,8 +6,43 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  for (const action of actions) {
+    const { type = null, extraData = {}, keysToRemove = [] } = action;
+
+    let newObj;
+
+    if (states.length === 0) {
+      newObj = { ...state };
+    } else {
+      newObj = { ...states[states.length - 1] };
```

Comment:
Instead of using `let` for `newObj`, it would be better to use `const` inside each case of the switch statement to avoid reassigning to `newObj`. This ensures that you're creating a new object for each action and not mutating the previous state or the new state object.